### PR TITLE
fix(cart): cart item state is set incorrectly in cart item reducers

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The cart item state reducers aren't correct, and sometimes even set "state" on the cart item entity instead of "daffState".

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Notes
The reason the tests were passing before is because the `DaffCartItemStateFactory` was setting the `daffState` incorrectly. I fixed the factory here.

### CODE REVIEW

File: libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts [hunk 1]
```diff
@@ -18,14 +18,20 @@ export function daffCartItemEntitiesReducer<
 	U extends DaffCartItemInput = DaffCartItemInput,
 	V extends DaffCart = DaffCart,
 >(
-  state = daffCartItemEntitiesAdapter<T>().getInitialState({ daffState: DaffCartItemStateEnum.Default }),
+  state = daffCartItemEntitiesAdapter<T>().getInitialState(),
```
Comment (griest024) [inline libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts:21]: Why did you remove this arg? Did it not set a default value for `daffState`?

File: libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts [hunk 2]
```diff
@@ -18,14 +18,20 @@ export function daffCartItemEntitiesReducer<
...
+			return adapter.addAll(action.payload.map(item => ({
+				...item,
+				daffState: DaffCartItemStateEnum.Default
+			})), state);
 		case DaffCartItemActionTypes.CartItemLoadSuccessAction:
-			return adapter.upsertOne(action.payload, state);
+			return adapter.upsertOne({
+				...action.payload,
+				daffState: DaffCartItemStateEnum.Default
+			}, state);
```
Comment (griest024) [inline libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts:34]: Are you sure we want to reset the state on item loads?

**Review by griest024 — CHANGES_REQUESTED** [review]

File: libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts (hunk 1 above)
Comment (lderrickable) [reply libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts:21]: Turns out it just did this:
```
{
    entities: {},
    ids: [],
    daffState: "default"
}
```
I replaced what this tried to do with setting `daffState: "default"` on each cart item on a `DaffCartLoadSuccess` and a `DaffCartItemListSuccess`

File: libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts (hunk 2 above)
Comment (lderrickable) [reply libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts:34]: I think it should be ok, because the `DaffCartLoad`, `DaffCartItemLoad`, and `DaffCartItemList` actions should only be called on initial state loading. I maybe could go without it, but the initial cart items daffState would be undefined until some kind of mutating action happened to them, which could be a problem if an app dev has a `daffState === "default"` check somewhere

Comment (griest024) [reply libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts:34]: > I think it should be ok, because the DaffCartLoad, DaffCartItemLoad, and DaffCartItemList actions should only be called on initial state loading

I think that largely depends on the app dev, we don't decide when this happens.

I know its more verbose, but I think `daffState: action.payload.daffState || DaffCartItemStateEnum.Default` is safer.

Comment (lderrickable) [reply libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts:34]: daffState will never exist on a DaffCartLoadSuccess.payload.items[], because it doesn't exist in the platform. 

`daffState: action.payload.daffState || DaffCartItemStateEnum.Default` will always result in `daffState: DaffCartItemStateEnum.Default`

Comment (griest024) [reply libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts:34]: oh yeah...so you would have to do something like:
```ts
adapter.upsertOne({
	...action.payload,
	daffState: state.entities[action.payload.id].daffState || DaffCartItemStateEnum.Default
}, state);
```

not great...a slightly more robust pattern that works for all default values would be something like:
```ts
const defaultStatefulCartItem: Partial<DaffStatefulCartItem> = {
  daffState: DaffCartItemStateEnum.Default,
}
...
adapter.upsertOne({
  ...defaultStatefulCartItem,
  ...state.entities[action.payload].id,
  ...action.payload,
}, state);
```

I'm not sure if its worth it, but I still don't really like the idea of resetting the state for immutable loads.

Comment (lderrickable) [reply libs/cart/state/src/reducers/cart-item-entities/cart-item-entities.reducer.ts]: `state.entities[action.payload.id].daffState || DaffCartItemStateEnum.Default` makes sense to me. Definitely better. Changed

**Review by griest024 — APPROVED** [review]
